libobs/obs-platform.h

```diff
@@ -0,0 +1,56 @@
+/******************************************************************************
+    Copyright (C) 2019 by Jason Francis <cycl0ps@tuta.io>
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+#pragma once
+
+#include "util/c99defs.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum obs_platform_type {
+	OBS_PLATFORM_DEFAULT,
```

Comment:
Id say dual name default X11 GLX, but this might not be everyone' taste.
Add X11 EGL

Reply:
Added

Reply:
I feel default is ambiguous (and in a world where wayland may one day before the heat death of the universe take over, it may not always be best to have default forever mean X11). If we dont want two names just call it `X11_GLX`

Reply:
Alright. I removed the default platform entirely, and made `obs-nix-platform` be Unix-specific.